pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -199,7 +199,7 @@ func (a *index) podWorkloadBuilder(
 	WorkloadServices krt.Collection[model.ServiceInfo],
 ) func(ctx krt.HandlerContext, p *v1.Pod) *model.WorkloadInfo {
 	return func(ctx krt.HandlerContext, p *v1.Pod) *model.WorkloadInfo {
-		if !IsPodRunning(p) || p.Spec.HostNetwork {
+		if (!IsPodRunning(p) && !IsPodPending(p)) || p.Spec.HostNetwork {
```

Comment:
I don't get why we need pending. If its 'pending' its not running so it cannot send traffic?

Is it just that this makes sure we have the workload before the pod starts instead of ~100ms after?

Reply:
i think we should support init container btw

Reply:
> if it's "pending" doesn't that mean the init container has not started? 

No, pod phase of "pending" means init container haven't finished.

So agree, probably not the right metric.

> A Pod cannot be Ready until all init containers have succeeded. The ports on an init container are not aggregated under a Service. A Pod that is initializing is in the Pending state but should have a condition Initialized set to false.

Reply:
TIL!

"but one or more of the containers has not been started"

So technically it may not have started yet, but probably good enough. We should make sure its assigned to a node though -- it will start in pending but no node